libs/coin-evm/src/api/gasTracker/ledger.ts

```diff
@@ -54,30 +54,40 @@ export const getGasOptions = async ({
       useEIP1559 ? "?display=eip1559" : ""
     }`,
   }).then(({ data }) => ({
-    low: new BigNumber(data.low),
-    medium: new BigNumber(data.medium),
-    high: new BigNumber(data.high),
-    next_base: new BigNumber(data.next_base),
+    /**
+     * Note: the explorer API is supposed to return integer values for each field,
+     * but since we can't guarantee that, better be safe than sorry
+     */
+    low: new BigNumber(data.low).integerValue(),
+    medium: new BigNumber(data.medium).integerValue(),
+    high: new BigNumber(data.high).integerValue(),
+    next_base: new BigNumber(data.next_base).integerValue(),
   }));
 
   const EIP1559_BASE_FEE_MULTIPLIER: number = getEnv("EIP1559_BASE_FEE_MULTIPLIER");
 
+  /**
+   * Since our use of BigNumber implies only using integers, we need to round up to
+   * integer value with `integerValue()` after doing some math
+   * (like multiplying with a possible float value in this case)
+   */
+
   if (useEIP1559) {
     return {
       slow: {
-        maxFeePerGas: next_base.times(EIP1559_BASE_FEE_MULTIPLIER).plus(low),
+        maxFeePerGas: next_base.times(EIP1559_BASE_FEE_MULTIPLIER).plus(low).integerValue(),
         maxPriorityFeePerGas: low,
         gasPrice: null,
         nextBaseFee: next_base,
       },
       medium: {
-        maxFeePerGas: next_base.times(EIP1559_BASE_FEE_MULTIPLIER).plus(medium),
+        maxFeePerGas: next_base.times(EIP1559_BASE_FEE_MULTIPLIER).plus(medium).integerValue(),
```

Comment:
Aren't we already multiplying an integer at this point ? next_base is an integer right ?

Reply:
NVM "EIP1559_BASE_FEE_MULTIPLIER" could be a float value as you pointed